llvm/lib/CodeGen/SelectionDAG/LegalizeVectorTypes.cpp

```diff
@@ -4429,6 +4435,22 @@ SDValue DAGTypeLegalizer::SplitVecOp_VECTOR_HISTOGRAM(SDNode *N) {
 //  Result Vector Widening
 //===----------------------------------------------------------------------===//
 
+void DAGTypeLegalizer::ReplaceOtherWidenResult(SDNode *N, SDNode *WidenNode,
+                                               unsigned WidenResNo) {
+  assert(N->getNumValues() == 2 && "expected node with two results");
+  unsigned OtherNo = 1 - WidenResNo;
+  EVT OtherVT = N->getValueType(OtherNo);
+  if (getTypeAction(OtherVT) == TargetLowering::TypeWidenVector) {
+    SetWidenedVector(SDValue(N, OtherNo), SDValue(WidenNode, OtherNo));
+  } else {
```

Comment:
This needs a check that the other operand is also a vector.

Reply:
Every `_UnaryOpWithTwoResults` method currently assumes both results are vectors (of the same element count), which has to hold for both `FSINCOS` and `FREXPR`. I'm not sure it should be further generalized while there are no nodes with mismatched values. Maybe the name could be changed to make this assumption explicit?

Reply:
I guess that's not really by design but rather because there are no operations that return a vector and a non-vector result. Perhaps you can just add an explicit assert that the type of `N`/`WidenNode` is a vector?

Reply:
Added an assert at the top of the function :+1: